lib/validators/stripped_length_validator.rb

```diff
@@ -3,14 +3,15 @@
 class StrippedLengthValidator < ActiveModel::EachValidator
   def self.validate(record, attribute, value, range)
     if !value.nil?
-      html_comments_regexp = /<!--(.*?)-->/
-      stripped_length = value.gsub(html_comments_regexp, '')
-      stripped_length = stripped_length.strip.length
+      value = value.dup
+      value.gsub!(/<!--(.*?)-->/, '') # strip HTML comments
+      value.gsub!(/:\w+(:\w+)?:/, "X") # replace emojis with a single character
+      value.gsub!(/\.{2,}/, '…') # replace multiple ... with …
+      value.gsub!(/\,{2,}/, ',') # replace multiple ,,, with ,
```

Comment:
I wonder if instead we could just remove any punctuation characters? I don't think a post with just punctuation characters should be valid? 

Something like

```ruby
   value.gsub!(/[[:punct:]]/, "")
```

Reply:
> I don't think a post with just punctuation characters should be valid?

Yes, but but not counting punctuation at all would lead to cases where seemingly valid sentences would be flagged as invalid.
E.g. "Is it 20 characters?" would be just under 20-character minimum, while being exactly 20 characters long.

Reply:
Good point. I kinda hate we have to re-implement Typographer here... Thinking out loud, can we somehow extract the Typographer?

Reply:
@CvX described exactly my thoughts and that problem is somewhat similar to the emoji one. We want them to count for something, but not their actual text. Typographer is implemented in JavaScript, deep in the Markdown pipeline and we need it here, on the Ruby side. Of course, we can use mini_racer, but given that this is a validation that may execute many times, I think we should keep it as fast as possible.

Reply:
@ZogStriP Technically, the typographer is now already extracted since #12417. It's "just" (heavy quotes there 😉) the matter of adding (and exporting) a function that would expose its functionality in a friendly way.